### PR TITLE
Filter exercises by description language

### LINKED ISSUE #1657: Searchable exercise description language
When choosing exercises to add to a series, it should be possible to filter exercises based on the language their description is available in. This requires saving the available languages in the database and allowing a language filter similar to filtering on tags and programming languages.

### PR DESCRIPTION
**Pull request by winniederidder**
This pull request adds the option to filter activities based on the languages their description is available in.

Image coming when implemented 
- [x] Tests were added
- [ ] Documentation update can be found at dodona-edu/dodona-edu.github.io#

Closes #1657  .

### CODE REVIEW

**Comment by winniederidder** [comment]
Visuals of filtered results:
filter_results
Current color of tag is simply gray

File: test/factories/exercises.rb [hunk 1]
```diff
@@ -29,6 +29,8 @@
     sequence(:name_nl) { |n| name || "Oefening #{n}" }
     sequence(:name_en) { |n| name || "Exercise #{n}" }
 
+    description_nl_present { false }
+    description_en_present { false }
```
Comment (rien) [inline test/factories/exercises.rb:33]: I would make this `true` for the `description_html_stubbed` and `description_md_stubbed` factory traits.

File: app/views/activities/index.html.erb [hunk 2]
```diff
@@ -5,7 +5,7 @@
         <h2 class="card-title-text"><%= t ".title" %></h2>
       </div>
       <div class="card-supporting-text">
-        <%= render partial: 'layouts/searchbar', locals: {labels: @labels, programming_languages: @programming_languages, repositories: @repositories, activity_types: [ContentPage, Exercise]} %>
+        <%= render partial: 'layouts/searchbar', locals: {labels: @labels, programming_languages: @programming_languages, repositories: @repositories, activity_types: [ContentPage, Exercise], description_languages: ["English", "Nederlands"]} %>
```
[the remark this answers is not in the document]
Comment (winniederidder) [reply app/views/activities/index.html.erb:8]: How would I go about that then? Use en, nl and add translation keys in the js locale files?

[the remark this answers is not in the document]
Comment (rien) [reply app/views/activities/index.html.erb:8]: That seems like a good solution.

**Comment by winniederidder** [comment]
Translated languages:
translated-languages

File: app/assets/javascripts/index.js [hunk 3]
```diff
@@ -254,7 +260,15 @@ function initFilterIndex(_baseUrl, eager, actions, doInitFilter, filterCollectio
             if (!collection) {
                 return false;
             }
-            return collection.data.map(el => el.name).includes(e.attrs.name);
+            let valid = collection.data.map(el => el.name).includes(e.attrs.name);
+            const newElementId = e.attrs.id.toString(); // ensure comparison is String-based
+            if (valid && collection.multi) {
```
Comment (bmesuere) [inline app/assets/javascripts/index.js:265]: can you add a comment to the code about what this is for?

**Comment by winniederidder** [comment]
series-filter
When filtering by language, only 3 pages appear instead of 10, similar to /activities, so I think this works.

**Comment by bmesuere** [comment]
the natural language should also be added to the table there

**Comment by winniederidder** [comment]
filter-by-language
Visuals with language in small, waiting for https://github.com/dodona-edu/example-exercises/pull/14 to be merged to also show labels

**Comment by winniederidder** [comment]
with-labels
Labels and languages in the table

**Comment by winniederidder** [comment]
reorder-tooltip
Tooltip and new column ordering

**Comment by winniederidder** [comment]
column-width
Added column xs-1 for the languages as we only support 2 languages right now.

**Pull request merged**